src/System.CommandLine/Argument.cs

```diff
@@ -116,9 +116,20 @@ bool DefaultConvert(SymbolResult symbol, out object value)
 
         public object GetDefaultValue() => _defaultValue?.Invoke();
 
-        public void SetDefaultValue(object value) => SetDefaultValue(() => value);
+        public void SetDefaultValue(object value)
+        {
+            if (value == null)
```

Comment:
Should we be adopting C# 8-isms with `is null`?
```suggestion
            if (value is null)
```